src/typings/types.ts

```diff
@@ -35,6 +35,11 @@ export type Pattern = RegExp | string;
 
 export type ReplaceType<O, K extends Prop, V> = Pick<O, Exclude<keyof O, K>> & { [p in K]: V };
 
+/**
+ * Obtain the values type of a array
+ */
+export type ArrayType<A extends any[]> = A extends Array<infer V> ? V : any;
```

Comment:
```suggestion
export type ArrayType<A extends any[]> = A extends Array<infer V> ? V : never;
```
To prevent usage for non-arrays

Reply:
thx!